### ISSUE TITLE
Add compiler flag to elide all trace-level (or higher) logging

### ISSUE THREAD

**Issue by sarahgerweck** [Log4s/log4s#1]
The suggestion came up today to add a compiler flag that would tell the compiler to just skip any statements of a certain level or lower. Something like `isTraceEnabled` is pretty cheap anyway, but compiling it out is even cheaper!

**Comment by lefou** [comment]
One legit reason why one would need such a feature is to hide some internal details from non-debug production code, which is not unusual in closed source projects. 
On the other hand, maybe tools like ProGuard already support such shrinking and do other optimizations and obfuscations as well. So, I think, this feature is not really needed.

**Comment by sarahgerweck** [comment]
Good point @lefou. I've treated this as a very low priority; do you have a use case for this?
